src/transformation/visitors/function.ts

```diff
@@ -142,7 +143,12 @@ export function transformFunctionBodyHeader(
 
     // Push spread operator here
     if (spreadIdentifier && isRestParameterReferenced(spreadIdentifier, bodyScope)) {
-        const spreadTable = wrapInTable(lua.createDotsLiteral());
+        let spreadTable: lua.Expression;
+        if (context.luaTarget === LuaTarget.Lua50) {
+            spreadTable = lua.createArgLiteral();
+        } else {
+            spreadTable = wrapInTable(lua.createDotsLiteral());
+        }
```

Comment:
```suggestion
        let spreadTable = context.luaTarget === LuaTarget.Lua50
            ? lua.createArgLiteral()
            : wrapInTable(lua.createDotsLiteral());
```

Reply:
shouldnt it be `const spreadTable`

Reply:
Yes it should, although I'm sure eslint would complained either way 😅

Reply:
Fixed with 2836859476950fb158ffb0dc4bff9fdefc5eeb24 (using `const`!).